### PR TITLE
feat: Add Create Expense Claim button in Delivery Trip

### PR DESCRIPTION
**Pull request by GangaManoj**
_Proposed Changes:_

- **Delivery Note:**
1.  Add a button that allows you to create an Expense Claim record from a Delivery Trip record, if the driver is an employee.
2. Add a new field called _Employee_ that displays the Employee Code for the driver.

Screenshot 2021-04-30 at 3 57 10 AM

- **Expense Claim:**
1. Add a new field called _Delivery Trip_ whose value is fetched from the Delivery Trip record from which the expense claim is created. This field will only be displayed if it's non-empty.
2. Fetch the Employee details from the Delivery Trip doc from which the Expense Claim record is created.

photo_2021-05-04 05 40 13

Docs: https://github.com/frappe/erpnext_documentation/pull/312

### CODE REVIEW

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js [hunk 1]
```diff
@@ -40,6 +40,17 @@ frappe.ui.form.on('Delivery Trip', {
 		})
 	},
 
+	onload: function (frm) {
+		// if (frm.doc.docstatus == 1) {
```
Comment (nextchamp-saqib) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.js]: Uncomment this?

**Review by nextchamp-saqib — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/expense_claim/expense_claim.json [hunk 2]
```diff
@@ -365,13 +366,19 @@
    "label": "Total Taxes and Charges",
    "options": "Company:company:default_currency",
    "read_only": 1
+  },
+  {
+   "fieldname": "delivery_trip",
```
Comment (nextchamp-saqib) [inline erpnext/hr/doctype/expense_claim/expense_claim.json:372]: Only show this field if it is set, keep it hidden if not set.

File: erpnext/stock/doctype/delivery_trip/delivery_trip.js (hunk 1 above)
Comment (GangaManoj) [reply erpnext/stock/doctype/delivery_trip/delivery_trip.js]: Done

File: erpnext/hr/doctype/expense_claim/expense_claim.json (hunk 2 above)
Comment (GangaManoj) [reply erpnext/hr/doctype/expense_claim/expense_claim.json:372]: Done

**Comment by nextchamp-saqib** [comment]
@GangaManoj  Just realized, the Employee field is not mandatory in Driver doctype, so there might be cases where Employee won't be fetched from Driver. In such cases, the Expense Claim button should be hidden.

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py [hunk 3]
```diff
@@ -394,3 +395,15 @@ def get_driver_email(driver):
...
 	email = frappe.db.get_value("Employee", employee, "prefered_email")
 	return {"email": email}
+
+@frappe.whitelist()
+def make_expense_claim(source_name, target_doc=None):
+	doc = get_mapped_doc("Delivery Trip", source_name,
+		{"Delivery Trip": {
+			"doctype": "Expense Claim",
+			"field_map": {
+				"delivery_trip": "name"
```
Comment (nabinhait) [inline erpnext/stock/doctype/delivery_trip/delivery_trip.py]:
```suggestion
				"name": "delivery_trip"
```

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/delivery_trip/delivery_trip.py (hunk 3 above)
Comment (GangaManoj) [reply erpnext/stock/doctype/delivery_trip/delivery_trip.py]: Done

File: erpnext/stock/doctype/delivery_trip/test_delivery_trip.py [hunk 4]
```diff
@@ -7,7 +7,8 @@
 
 import erpnext
 import frappe
-from erpnext.stock.doctype.delivery_trip.delivery_trip import get_contact_and_address, notify_customers
+from erpnext.stock.doctype.delivery_trip.delivery_trip import get_contact_and_address, notify_customers, make_expense_claim
+from erpnext.hr.doctype.employee.test_employee import make_employee
```
Comment (nextchamp-saqib) [inline erpnext/stock/doctype/delivery_trip/test_delivery_trip.py]:
```suggestion
```

**Comment by nabinhait** [comment]
@GangaManoj Why 16 commits are required for this feature?

**Comment by GangaManoj** [comment]
@nabinhait It started out with 3 and additional commits were made to correspond to each suggestion that was made during the review process + to fix Sider issues. I thought I should make separate ones for each, since they all serve different purposes. Is that a problem? Is there a convention I should be following?

**Pull request merged**